src/Numerics/DGmethods/NumericalFluxes.jl

```diff
@@ -272,39 +239,33 @@ struct CentralNumericalFluxFirstOrder <: NumericalFluxFirstOrder end
 function numerical_flux_first_order!(
     ::CentralNumericalFluxFirstOrder,
     balance_law::BalanceLaw,
-    fluxᵀn::Vars{S},
+    states::NamedTuple,
     normal_vector::SVector,
-    state_conservative⁻::Vars{S},
-    state_auxiliary⁻::Vars{A},
-    state_conservative⁺::Vars{S},
-    state_auxiliary⁺::Vars{A},
     t,
-) where {S, A}
+)
 
-    FT = eltype(fluxᵀn)
+    FT = eltype(states.flux)
     num_state_conservative = number_state_conservative(balance_law, FT)
-    fluxᵀn = parent(fluxᵀn)
 
-    flux⁻ = similar(fluxᵀn, Size(3, num_state_conservative))
+    flux⁻ = similar(states.flux, Size(3, num_state_conservative))
```

Comment:
The `similar` call here fails because it doesn't properly guess the type of `states.flux`